src/Lobby/PlayerList.js

```diff
@@ -18,25 +18,33 @@ const PlayerList = (props) => {
     setTotalPlayers(playerNames);
   };
   useEffect(() => {
-    retrievePlayer();
+    fetchPlayerNames();
 
     let mySubscription = supabase
       .from("Players")
       .on("UPDATE", (payload) => {
-        if (payload.new.roomID === roomCode) {
+        if (
+          payload.new.gameStatus === null &&
+          payload.new.roomID === roomCode
+        ) {
           setTotalPlayers((totalPlayers) =>
             totalPlayers.concat(payload.new.name)
           );
         }
       })
       .subscribe();
-  }, []);
+    if (hasGameStarted === true) {
+      supabase.removeSubscription(mySubscription);
+    }
+  }, [hasGameStarted]);
```

Comment:
Lets use the cleanup functionality built into useEffect instead. https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1

Reply:
It's likely the component is unmounted before it even sees `hasGameStarted` change to `true`